state/mongodb/mongodb_query.go

```diff
@@ -0,0 +1,176 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package mongodb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/dapr/components-contrib/state"
+	"github.com/dapr/components-contrib/state/query"
+)
+
+type Query struct {
+	query  string
+	filter interface{}
+	opts   *options.FindOptions
+}
+
+func (q *Query) VisitEQ(f *query.EQ) (string, error) {
+	// { <key>: <val> }
+	return fmt.Sprintf("{ %q: %q }", f.Key, f.Val), nil
```

Comment:
`key` or `value` could have a content that could turn this string into an invalid JSON. Instead, build the query object directly in the visitor methods.